ui/src/op_effects.rs

```diff
@@ -0,0 +1,298 @@
+/// Maps an opcode and returns a vector of named affected indices
+pub fn stack_indices_affected(op: u8) -> Vec<(usize, &'static str)> {
+    match op {
```

Comment:
nit: can minimize this by using `_ => vec![]`?

Reply:
Ye i can - the reason i did this was to make it explicit if opcodes get added. happy to switch

Reply:
the `op` will be compared against each match arm, 
not a big deal but should consider rearranging because match arms are executed consecutively, however llvm will probably optimize this to some degree

Reply:
no this would be compiled into a jump table so it will be constant time